### ISSUE TITLE
ptest infrastructure is unnecessary?

### ISSUE THREAD

**Issue by RAOF** [MirServer/mir#2850]
We have a bunch of infrastructure to support a `make ptest` target for running the Mir test suite in parallel. It turns out that `ctest` has its own, built in, support for running the test suite in parallel with the `-j` option (not to be confused with `make -j`; you need to get `make` to pass this through with `make ARGS+=-j9`) or the `CTEST_PARALLEL_LEVEL` environment variable.

Do we actually need the ptest infrastructure? Can it be removed?

**Comment by Saviq** [comment]
+1, the whole testing setup feels like it could be re-done from scratch. I'm certain there's simpler ways achieving (nearly) the same result with existing utilities.

We'd need someone to write down what it does today, to be able to compare, though.

**Comment by AlanGriffiths** [comment]
Doesn't `ptest` already use `ctest -j`?

IIRC the "infrastructure" is to split out the tests into sensible sized chunks (i.e. treating each fixture as a "chunk")

(Not saying it can't be revisited, just clarifying)

**Comment by Saviq** [comment]
> IIRC the "infrastructure" is to split out the tests into sensible sized chunks (i.e. treating each fixture as a "chunk")

Yes, IIUC it discovers all the suites and uses separate `add_test()` for each so that `ctest -j` can then parallelize better. Maybe we can split the test executables along those lines, rather than relying on `--gtest_filter`. Or maybe cmake it in a way that there's a separate `add_test()` registered in the first place.
